webcompat/static/js/lib/mixins/pagination.js

```diff
@@ -0,0 +1,162 @@
+*/
+
+issueList.PaginationControlsView = Backbone.View.extend({
+  initialize: function(options) {
+    this.el = options.el;
+  },
+  events: {
+    'click .js-pagination-previous': 'broadcastPrevious',
+    'click .js-pagination-next': 'broadcastNext',
+  },
+  broadcastNext: function(e) {
+    issueList.events.trigger('paginate:next', e);
+    e.preventDefault();
+  },
+  broadcastPrevious: function(e) {
+    issueList.events.trigger('paginate:previous', e);
+    e.preventDefault();
+  }
+});
+
+/*
+Pagination Mixin Usage:
+issueList.IssueView = Backbone.View.extend(
+  _.extend({}, PaginationMixin, {
+  // regular view code here
+  // ...
+});
+
+The mixin is initialized like so:
+
+PaginationMixin.initMixin(hostView, hostModel, parentContainerEl);
+
+parentContainerEl: a jQuery object that refers to a parent container of both
+the hostView *and* the pagination controls.
+
+The PaginationMixin requires the "host view" to
+implement the following two methods:
+
+fetchAndRenderIssues()
+updateModelParams()
+
+Check out issueList.IssueView for an example.
+*/
+
+function PaginationMixin() {
+  this.initMixin = function(hostView, hostModel, parentContainerEl) {
+    this.view = hostView;
+    this.model = hostModel;
+    this.parentContainerEl = parentContainerEl;
+
+    this.paginationControls = new issueList.PaginationControlsView(
+      {el: this.parentContainerEl}
+    );
+
+    issueList.events.on('paginate:next', _.bind(this.requestNextPage, this));
+    issueList.events.on('paginate:previous', _.bind(this.requestPreviousPage, this));
+  };
+
+  this.initPaginationLinks = function(issuesCollection) {
+    // if either the next or previous page numbers are null
+    // disable the buttons and add .is-disabled classes.
+    var nextButton = this.paginationControls.el.find('.js-pagination-next');
+    var prevButton = this.paginationControls.el.find('.js-pagination-previous');
+    var nextPage = issuesCollection.getNextPage();
+    var prevPage = issuesCollection.getPrevPage();
+    var isLastPage = function() {
+      return nextPage == null;
+    };
+    var isFirstPage = function() {
+      return prevPage == null;
+    };
+    var isSinglePage = isLastPage() && isFirstPage();
+
+    if (!issuesCollection.length || isSinglePage) {
+      // hide pagination buttons if there are no results,
+      // or the results are limited to a single page.
+      nextButton.addClass('wc-hidden');
+      prevButton.addClass('wc-hidden');
+      return;
+    }
+
+    nextButton.removeClass('wc-hidden')
+              .prop('disabled', isLastPage())
+              .toggleClass('is-disabled', isLastPage());
+    prevButton.removeClass('wc-hidden')
+              .prop('disabled', isFirstPage())
+              .toggleClass('is-disabled', isFirstPage());
+
+    if (nextPage) {
+      // chop off leading "/api" and set @href
+      nextButton.attr('href', issuesCollection.getNextPage().slice(4));
```

Comment:
I'd prefer .substring rather than .slice to make it (more) obvious we're dealing with a string and not an array.. but maybe that's just me :-p